controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -599,6 +603,85 @@ func TestKubeadmControlPlaneReconciler_adoption(t *testing.T) {
 	})
 }
 
+func TestKubeadmControlPlaneReconciler_SetInfraRefAnnotationToMachines(t *testing.T) {
```

Comment:
Can we use ginkgo for new tests instead of plain unit tests?

Reply:
Okay, I didn't know that we were transitioning to gingko tests.

Reply:
We'll need to add a policy in the contributing guide soon, just wanted to get started on it given that we're adding new functionality